File: app/src/main/java/fr/clerc/tp2/MainActivity.java
package fr.clerc.tp2;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.LinearLayout;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

public class MainActivity extends AppCompatActivity {

    RecyclerView recyclerView;
    RecyclerViewAdapter recyclerAdapter;

    public static final String STUDENT_ID = "student_id";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerView);

        //Initialize layoutManager with a LinearLayoutManager, by default vertical
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        //Add list divider between each line
        DividerItemDecoration dividerItemDecoration = new DividerItemDecoration(recyclerView.getContext(), LinearLayout.VERTICAL);
        recyclerView.addItemDecoration(dividerItemDecoration);

        //Create adapter with the OnItemClickListener implementation
        recyclerAdapter = new RecyclerViewAdapter(StudentContent.ITEMS, new RecyclerViewAdapter.OnItemClickListener() {
            @Override
            public void onItemClick(View view, int position) {
                Student item = recyclerAdapter.getItem(position);

                Intent intent = new Intent(getApplicationContext(), StudentDetailsActivity.class);
                intent.putExtra(STUDENT_ID, item.id);
                startActivity(intent);
            }
        });

        //Set the adapter
        recyclerView.setAdapter(recyclerAdapter);
    }
}

File: app/src/main/java/fr/clerc/tp2/kotlin/MainActivity.kt
package fr.clerc.tp2.kotlin

import android.content.Intent
import android.os.Bundle
import android.widget.LinearLayout.VERTICAL
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import fr.clerc.tp2.R
import kotlinx.android.synthetic.main.activity_main.recyclerView

class MainActivity : AppCompatActivity() {

    companion object {
        const val STUDENT_ID: String = "student_id"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)

        recyclerView.layoutManager = LinearLayoutManager(this)

        val itemDecoration = DividerItemDecoration(this, VERTICAL)
        recyclerView.addItemDecoration(itemDecoration)

        val adapter = RecyclerViewAdapter(StudentContent.ITEMS) { student ->
            val intent = Intent(this, StudentDetailsActivity::class.java)
                .apply { putExtra(STUDENT_ID, student.id) }
            startActivity(intent)
        }

        recyclerView.adapter = adapter
    }
}